xarray_sentinel/sentinel1.py

```diff
@@ -375,18 +377,61 @@ def open_pol_dataset(
     return xr.Dataset(attrs=attrs, data_vars={"measurement": arr})
 
 
-def crop_burst_dataset(pol_dataset: xr.Dataset, burst_index: int) -> xr.Dataset:
-    if burst_index < 0 or burst_index >= pol_dataset.attrs["number_of_bursts"]:
-        raise IndexError(f"{burst_index=} out of bounds")
+def find_bursts_index(
+    pol_dataset: xr.Dataset,
+    azimuth_anx_time: float,
+    use_center: bool = False,
+) -> int:
+    lines_per_burst = pol_dataset.attrs["lines_per_burst"]
+    anx_datetime = pol_dataset.attrs["sat_anx_datetime"]
+    anx_datetime = np.datetime64(anx_datetime)
+    azimuth_anx_time = pd.Timedelta(azimuth_anx_time, unit="s")
+    if use_center:
+        azimuth_anx_time_center = (
+            pol_dataset.azimuth_time[lines_per_burst // 2 :: lines_per_burst]
+            - anx_datetime
+        )
+        distance = abs(azimuth_anx_time_center - azimuth_anx_time)
+    else:
+        azimuth_anx_time_first_line = (
+            pol_dataset.azimuth_time[::lines_per_burst] - anx_datetime
+        )
+        distance = abs(azimuth_anx_time_first_line - azimuth_anx_time)
+    return distance.argmin().item()  # type: ignore
+
+
+def crop_burst_dataset(
+    pol_dataset: xr.Dataset,
+    index: T.Optional[int] = None,
+    azimuth_anx_time: T.Optional[float] = None,
+    use_center: bool = False,
+) -> xr.Dataset:
+
+    if (index is not None) and (azimuth_anx_time is not None):
+        raise ValueError(
+            "only one keyword between 'index' and 'azimuth_anx_time' must be defined"
+        )
+
+    if index is None:
+        if azimuth_anx_time is not None:
+            index = find_bursts_index(
+                pol_dataset, azimuth_anx_time, use_center=use_center
+            )
+        else:
+            raise ValueError(
+                "one keyword between 'index' and 'azimuth_anx_time' must be defined"
+            )
+
+    if index < 0 or index >= pol_dataset.attrs["number_of_bursts"]:
+        raise IndexError(f"{index=} out of bounds")
 
     lines_per_burst = pol_dataset.attrs["lines_per_burst"]
     ds = pol_dataset.sel(
-        line=slice(
-            lines_per_burst * burst_index, lines_per_burst * (burst_index + 1) - 1
-        )
+        line=slice(lines_per_burst * index, lines_per_burst * (index + 1) - 1)
     )
+
     ds = ds.swap_dims({"line": "azimuth_time", "pixel": "slant_range_time"})
-    ds.attrs["burst_index"] = burst_index
+    ds.attrs["burst_index"] = index
 
```

Comment:
something like (untested):

```suggestion
ds.attrs["azimuth_anx_time_first_line"] = (ds.azimuth_time - anx_time).seconds()
```